src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs

```diff
@@ -695,9 +695,20 @@ public void OnTrailersComplete()
                 // (DisposeContext logs StatusCode).
                 SetBadRequestState(ex);
                 ReportApplicationError(ex);
-            }
+            }            
             catch (Exception ex)
             {
+                if (ex is OperationCanceledException && RequestAborted.IsCancellationRequested)
+                {
+                    Log.RequestAborted();
+
+                    if (!HasResponseStarted)
+                    {
+                        StatusCode = 499;
+                    }
+
+                    return;
+                }
                 ReportApplicationError(ex);
```

Comment:
I have the same exact feedback as I do for `IISHttpContext`. We should also avoid the `RequestAborted` getter because in the uncancelled case that can unnecessarily lazily create a `CancellationTokenSource`.

```suggestion
                if (ex is OperationCanceledException && _connectionAborted)
                {
                    Log.RequestAborted();
                }
                else
                {
                    ReportApplicationError(ex);
                }
```

I also have the same feedback about changing the setting of the status code below where we have:

```csharp
                    // If the request was aborted and no response was sent, there's no
                    // meaningful status code to log.
                    StatusCode = 0;
````

Changing the control flow in `HttpProtocol` is even more dangerous than in `IISHttpContext` because stuff like `await FireOnCompleted();` and `application.DisposeContext(context, _applicationException);` which absolutely should still run in this case are not in a finally. This is important for Hosting's request completed logs and running `HttpResposne.OnCompleted` callbacks. We have tests that should have caught this, but none of them throw an OperationCanceledException so it slipped by.

https://github.com/dotnet/aspnetcore/blob/249dfcb2bc5dd30152ce10c47e8c77f2cc95ca81/src/Servers/Kestrel/test/InMemory.FunctionalTests/ResponseTests.cs#L300-L313

We should add a variant of that test that calls `context.RequestAborted.ThrowIfCancellationRequested();` after `context.Abort();` instead of `throw new Exception()` and verifies we get 499 as the status code. Then we should update the existing tests to expect 499 instead of 0. This is of course assuming we decide we want to switch from using 0 to 499 for this purpose.

I personally think 0 is probably sufficient for this case, but if 499 is widely used for this, I can understand changing it. Hopefully it doesn't break anything.